File: day 1/part1.hs
import System.IO
import Data.List (sort)

part1 :: String -> Int
part1 inp = sum [abs ((read x :: Int) - (read y :: Int)) | (x, y) <- zip (sort left) (sort right)]
    where
        gen_lists [] = ([], [])
        gen_lists (xs:xss) = (head two_words:left, head (tail two_words):right)
            where
                (left, right) = gen_lists xss
                two_words = words xs
        
        (left, right) = gen_lists (lines inp)
        

main :: IO ()
main = do
    inp <- readFile "input.txt"
    print (part1 inp)

File: day 1/part1.py
from aocd import get_data
data = get_data(day=1, year=2024)

left, right = [], []
for line in data.split('\n'):
    l, r = line.split()
    left.append(int(l))
    right.append(int(r))

left.sort()
right.sort()

total = 0
for i in range(len(left)):
    total += abs(right[i] - left[i])

print(total)